smoke-tests/spec/master_nodes_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "spec_helper"
+
+describe "master nodes", speed: "fast" do
+  # normalise pod names for ease of comparison, e.g.
+  #
+  #   calico-node-mv48v -> calico-node
+  #   etcd-manager-events-ip-172-20-100-1.eu-west-2.compute.internal -> etcd-manager-events
+  #
+  def shorten_pod_name(name, node_name)
+    name
+      .sub(node_name, "") # remove node name
+      .sub(/-[^-]*$/, "") # remove last '-' and everything after it
+  end
+
+  # return a list of [namespace, shortened-pod-name] tuples, for
+  # every pod running on a given node
+  def pods_on_master(pods, node_name)
+    node_pods = pods.find_all { |p| p.dig("spec", "nodeName") == node_name }
+    node_pods.map { |p|
+      [
+        p.dig("metadata", "namespace"),
+        shorten_pod_name(p.dig("metadata", "name"), node_name),
+      ]
+    } .sort
+  end
+
+  let(:masters) {
+    kubectl_items("get nodes")
+      .find_all { |node| node.dig("metadata", "labels", "kubernetes.io/role") == "master" }
+  }
+
+  let(:pods) { kubectl_items("get pods --all-namespaces") }
+
+  # We should have 3 masters
+  specify do
+    expect(masters.size).to eq(3)
+  end
+
+  # Every master should have these pods running, in these namespaces
+  it "has standard pods" do
+    expected_pods = [
+      ["kube-system", "calico-node"],
```

Comment:
You could also add calico-kube-controller.

Reply:
Only appears as a single pod however.

Reply:
There is only one of those, but it doesn't run on a master node (necessarily? it's running on a worker node right now - could it run on a master?)